src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,32 +43,50 @@ export const TodoModal: React.FC = () => {
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${selectedTodo.id}`}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
               type="button"
               className="delete"
               data-cy="modal-close"
+              onClick={() => {
+                setSelectedTodo(null);
+              }}
             />
           </header>
 
           <div className="modal-card-body">
             <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
+              {selectedTodo.title}
             </p>
 
             <p className="block" data-cy="modal-user">
               {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
+              {selectedTodo.completed ? (
+                <strong className="has-text-success">Done</strong>
+              ) : (
+                <strong className="has-text-danger">Planned</strong>
+              )}
 
               {' by '}
 
-              <a href="mailto:Sincere@april.biz">
-                Leanne Graham
-              </a>
+              {!hasError
+                ? (
+                  <a href={`mailto:${currentUser?.email}`}>
+                    {currentUser?.name}
+                  </a>
+                ) : (
+                  <strong>Unknown</strong>
+                )}
             </p>
+
+            {hasError && (
+              <div className="notification is-warning">
+                Server error!
+              </div>
```

Comment:
the cases same the same condition, so if we hasError we show unknown & server error is it expected?